model/src/activity/encrypted.rs

```diff
@@ -0,0 +1,143 @@
+use crate::activity::{ExeScriptCommand, ExeScriptCommandResult};
+use rand::Rng as _;
+use secp256k1::ecdh::SharedSecret;
+use secp256k1::{PublicKey, SecretKey};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Request {
+    pub activity_id: String,
+    pub batch_id: String,
+    pub timeout: Option<f32>,
+    pub command: RequestCommand,
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum RequestCommand {
+    Exec { exe_script: Vec<ExeScriptCommand> },
+    GetExecBatchResults { command_index: Option<usize> },
+}
```

Comment:
[note to self] to be implemented in https://github.com/golemfactory/ya-client/pull/33